.gitlab-ci.yml

```diff
@@ -1,37 +1,592 @@
+      mv core-image-full-cmdline-vexpress-qemu.ext4 integration/tests
+    - tar -xf host-tools.tar ./mender-artifact && mv mender-artifact /usr/local/bin/
+    - tar -xf host-tools.tar ./mender-stress-test-client && mv mender-stress-test-client /usr/local/bin/
+    - tar -xf host-tools.tar ./directory-artifact-gen && mv directory-artifact-gen /usr/local/bin/
   script:
-    - echo "Build parameters:"
-    - echo "DEPLOYMENTS_REV=${DEPLOYMENTS_REV}"
-    - echo "DEVICEAUTH_REV=${DEVICEAUTH_REV}"
-    - echo "GUI_REV=${GUI_REV}"
-    - echo "MENDER_REV=${MENDER_REV}"
-    - echo "MENDER_API_GATEWAY_DOCKER_REV=${MENDER_API_GATEWAY_DOCKER_REV}"
-    - echo "MENDER_ARTIFACT_REV=${MENDER_ARTIFACT_REV}"
-    - echo "MENDER_CLI_REV=${MENDER_CLI_REV}"
-    - echo "MENDER_CONDUCTOR_REV=${MENDER_CONDUCTOR_REV}"
-    - echo "MENDER_CONDUCTOR_ENTERPRISE_REV=${MENDER_CONDUCTOR_ENTERPRISE_REV}"
-    - echo "USERADM_REV=${USERADM_REV}"
-    - echo "INTEGRATION_REV=${INTEGRATION_REV}"
-    - echo "BASE_BRANCH=${BASE_BRANCH}"
-    - echo "RUN_INTEGRATION_TESTS=${RUN_INTEGRATION_TESTS}"
-    - echo "INVENTORY_REV=${INVENTORY_REV}"
-    - echo "BUILD_QEMUX86_64_UEFI_GRUB=${BUILD_QEMUX86_64_UEFI_GRUB}"
-    - echo "TEST_QEMUX86_64_UEFI_GRUB=${TEST_QEMUX86_64_UEFI_GRUB}"
-    - echo "BUILD_QEMUX86_64_BIOS_GRUB=${BUILD_QEMUX86_64_BIOS_GRUB}"
-    - echo "TEST_QEMUX86_64_BIOS_GRUB=${TEST_QEMUX86_64_BIOS_GRUB}"
-    - echo "BUILD_QEMUX86_64_BIOS_GRUB_GPT=${BUILD_QEMUX86_64_BIOS_GRUB_GPT}"
-    - echo "TEST_QEMUX86_64_BIOS_GRUB_GPT=${TEST_QEMUX86_64_BIOS_GRUB_GPT}"
-    - echo "BUILD_VEXPRESS_QEMU=${BUILD_VEXPRESS_QEMU}"
-    - echo "TEST_VEXPRESS_QEMU=${TEST_VEXPRESS_QEMU}"
-    - echo "BUILD_VEXPRESS_QEMU_FLASH=${BUILD_VEXPRESS_QEMU_FLASH}"
-    - echo "TEST_VEXPRESS_QEMU_FLASH=${TEST_VEXPRESS_QEMU_FLASH}"
-    - echo "BUILD_VEXPRESS_QEMU_UBOOT_UEFI_GRUB=${BUILD_VEXPRESS_QEMU_UBOOT_UEFI_GRUB}"
-    - echo "TEST_VEXPRESS_QEMU_UBOOT_UEFI_GRUB=${TEST_VEXPRESS_QEMU_UBOOT_UEFI_GRUB}"
-    - echo "BUILD_BEAGLEBONEBLACK=${BUILD_BEAGLEBONEBLACK}"
+    - source ${CI_PROJECT_DIR}/build_revisions.env
+    - export $(cat ${CI_PROJECT_DIR}/build_revisions.env | cut -d= -f1)
+    - cd integration/tests
+    - ./run.sh --no-download --machine-name $MACHINE_NAME
+
+test_integration_qemux86_64_uefi_grub:
+  stage: test
+  <<: *test_integration
+  dependencies:
+    - init_workspace
+    - build_servers
+    - build_qemux86_64_uefi_grub
+  variables:
+    MACHINE_NAME: qemux86-64
+  after_script:
+    - cp ${CI_PROJECT_DIR}/integration/tests/results.xml results_qemux86_64_uefi_grub.xml
+    - cp ${CI_PROJECT_DIR}/integration/tests/report.html report_qemux86_64_uefi_grub.html
+  artifacts:
+    expire_in: 2w
+    when: always
+    paths:
+      - results_qemux86_64_uefi_grub.xml
+      - report_qemux86_64_uefi_grub.html
+    reports:
+      junit: results_qemux86_64_uefi_grub.xml
+  only:
+    variables:
+      # TODO: investigate why the expressions doesn't work in GitLab
```

Comment:
apparently this was only recently merged: https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/27925 - should be part of the 12.0 or 12.1 release...

what might work (according to https://gitlab.com/gitlab-org/gitlab-ce/issues/27818#note_38463332): 
```
only:
  variables:
    - $RUN_INTEGRATION_TESTS == "true"
  variables:
    - $TEST_QEMUX86_64_UEFI_GRUB == "true"
```

Reply:
Thanks! Added it and testing tonight...

Reply:
It works :+1: